### DISCUSSION TITLE
Surface Duo Camera Stock Port

### DISCUSSION THREAD

**Discussion by Duonab** [Archfx/duo-de#109] [General]
Hello,
I couldn't find my old github account about a previous problem that keep happening related to the camera usage on the Duo DE Rom

Basically when you try to record horizontally or vertically with the third party camera that come with Duo De it either crash or export in a inverted picture format
Instead of being properly horizontally, the video is totally upside down
I wonder if this happen to anyone else but weirdly this problem doesn't exist on the stock version or there is a kind of issues with the port version of Android

Archfx, can you please take a look at what is happening here and also any possibility to port the stock version of the camera from 12L to the 15?

Thank you

**Comment by Archfx** [comment]
Have you enabled the [PHH settings](https://github.com/Archfx/duo-de/discussions/81) related to the camera?

**Reply by Duonab** [reply]
Hello,

I've done every possible things but problem still remain
When you make a video or take pictures
The contents is upside down in the gallery

**Reply by Archfx** [reply]
First Duo-De does not come with a third-party camera (specifically this build doesn't come with any third-party apps by default). The available camera app is from Google (AOSP) and the issues you mention here are not reproducible. This could potentially be a hardware issue in your particular unit.

**Reply by Duonab** [reply]
Yeah I do know that the camera App is the one from AOSP which is also a third party one in a way.

But as I said this problem doesn't exist on 12L stock
Which I don't think the problem belong to an hardware issue

I will check the make a video to show the problem

Basically just open the base camera that come with AOSP
On a original Surface Duo not Duo 2
Turn the phone horizontally and start to record a video or take picture
You will see that the saved video or picture is upside down
Even when you rotate the phone with the base camera something is odd about the viewfinder preview

**Reply by Archfx** [reply]
I think I know what is happening here, in the AOSP camera app click on the three dots, and on the menu you can flip the camera using the last icon.  

Just a clarification AOSP stands for Android Open Source Project.  DUO-DE is based on AOSP, therefore all the apps and services you see here are from Android itself not from a third-party developer.

**Reply by Duonab** [reply]
As I said I've done everything, something is wrong about how AOSP is dealing with the duo 1 camera process. I will check more closely into this
But as I said the problem doesn't exist on the Stock, so It is a software issue

Is not about the flip camera there is a issue about how the software part
Whether you use inverted setting to save the files in a correct ratio or not
It doesn't change anything

Basically the software doesn't know how to deal with the rotation of the phone and camera position

Try yourself don't use it as a selfie but use the left screen for the preview and the front camera on the right camera as a back camera then put it in horizontal like you want to film or take picture of a paysage

You'll see that the rotation is incorrect and the saving files is upside down
